autoscheduler/frontend/src/components/SchedulingPage/SchedulePreview/MiniSchedule/MiniSchedule.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import Meeting from '../../../../types/Meeting';
+import { LAST_HOUR, FIRST_HOUR } from '../../../../timeUtil';
+import DayOfWeek from '../../../../types/DayOfWeek';
+import * as parentStyles from '../../Schedule/Schedule.css';
+import * as styles from './MiniSchedule.css';
+import colors from '../../Schedule/meetingColors';
+
+interface MiniScheduleProps {
+  schedule: Meeting[];
+}
+
+const MiniSchedule: React.FC<MiniScheduleProps> = ({ schedule }) => {
+  // these must be unique because of how they're used below
+  const DAYS_OF_WEEK = ['M', 'T', 'W', 'R', 'F'];
+
+  // build rows from first and last hour
+  const HOURS_OF_DAY = [];
```

Comment:
These are constants, so move their definitions outside of the component so that only have to be calculated once